### DISCUSSION TITLE
Import advice

### DISCUSSION THREAD

**Discussion by shinypb** [cue-lang/cue#1431] [General] [answered]
[This page](https://cuelang.org/docs/tutorials/tour/packages/imports/) says ”it is good style to use the factored import statement”, but doesn't offer any information about why that is good style, or what the differences between factored and non-factored import statements are.

**Comment by myitcv** [comment]
@shinypb - I've moved your issue from https://github.com/cue-lang/cuelang.org ultimately to be a discussion as part of the main CUE repo.

**Comment by verdverm** [comment, marked as answer]
Factored is the parenthesis version, unfactored is the import statement per line. The term comes from Tour of Go and essentially means the token "import" has been factored out, in analogy to the mathematical concept.

https://stackoverflow.com/questions/59232785/why-is-it-called-a-factored-import-statement-in-golang

**Reply by myitcv** [reply]
Just building on the answer from @verdverm. The term can be seen here:

https://go.dev/tour/basics/2

Factored here is more a mathematical term. 

Per the [Go spec](https://go.dev/ref/spec#Constant_declarations), such an import declaration is referred to as a list, where the import keyword is "factored" out.
